module2/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,65 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, height>, width>& uncompressed) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(height * width);
+
+    std::for_each(uncompressed.begin(), uncompressed.end(), [&](auto row) {
+        auto groupBegin = row.begin();
+        auto nextGroupBegin = row.end();
+
+        do {
+            uint8_t firstPixelInGroup = *groupBegin;
+            nextGroupBegin =
+                std::find_if_not(groupBegin,
+                                 row.end(),
+                                 [&](auto currentPixel) {
```

Comment:
In capture list you should write explicit name of variable which you are going to reference in lambda